git/git_for_developers.tex

```diff
@@ -0,0 +1,136 @@
+\begin{itemize}
+    \item \textbf{Keep PRs Small and Focused:} Encourage contributions that are easy to review and discuss. Smaller changes are easier to understand and less likely to introduce errors.
+    \item \textbf{Use a Checklist:} Develop a review checklist to ensure consistency and thoroughness. This can include code style, testing, documentation, and performance considerations.
+    \item \textbf{Automated Checks:} Utilize GitHub Actions to run automated tests, linting, and other checks when PRs are opened or updated.
+\end{itemize}
+\end{frame}
+\begin{frame}{GitHub Actions for Automation}
+\begin{itemize}
+    \item \textbf{Event-Driven:} Trigger workflows on GitHub events like push, pull requests, or issue comments.
+    \item \textbf{Workflows and Actions:} Combine multiple actions to create workflows defined in YAML files.
+    \item \textbf{Hosted Runners:} Run workflows on GitHub-hosted runners or self-hosted runners.
+    \item \textbf{Examples:}
+    \begin{itemize}
+        \item \textbf{Automating Testing and Deployment}: Automatically run tests on every pull request or push to a specific branch. Deploy your application when a pull request is merged into the main branch.
+        \item \textbf{Custom Workflows for Project Management:} Auto-assign project issues to members based on labels. Auto-label pull requests based on modified file paths.
+    \end{itemize}
+\end{itemize}
+\end{frame}
+\begin{frame}{Securing Your Project}
+\begin{itemize}
+    \item \textbf{Branch Protection Rules:} Configure rules to protect branches, requiring pull request reviews, status checks before merging, and more.
+    \item \textbf{Dependabot:} Automatically scans your dependencies for known vulnerabilities and suggests updates or patches.
+    \item \textbf{Secret Scanning:} Detects secrets and credentials exposed in your code and provides alerts.
+    \item \textbf{Code Scanning:} Automatically scans your code for vulnerabilities when you push code to GitHub.
+\end{itemize}
+\end{frame}
+\begin{frame}{Extras}
+\begin{itemize}
+    \item \textbf{Interactive Rebase:}
+    \begin{itemize}
+        \item \textbf{Definition:} Tool for rewriting history, used to edit, delete, or squash commits.
+        \item \textbf{Use Case:} Cleaning up a feature branch before merging it into the main branch.
+    \end{itemize}
+    \item \textbf{Cherry-picking:}
+    \begin{itemize}
+        \item \textbf{Definition:} Allows you to pick a commit from one branch and apply it to another.
+        \item \textbf{Use Case:} Applying a bug fix from one branch to another without merging all changes. 
+    \end{itemize}
+    \item \textbf{Stashing Changes:}
+    \begin{itemize}
+        \item \textbf{Definition:} Temporarily, shelves (or stashes) change so you can work on a different task.
+        \item \textbf{Use Case:} Switching between branches without committing half-done work.
+    \end{itemize}
+\end{itemize}
+\end{frame}
```

Comment:
I would suggest breaking each of these into their own slides with their own graphics.  Also, consider an activity to reinforce these.

Reply:
I am considering removing one of the slides from the presentation because only one out of the three slides seems relevant to teamwork. Rebase damages project history, and if one felt the need to stash, the work should've probably been two branches anyway. Also, the slides are already quite lengthy. 

However, I have included a slide/demo on cherry-picking for now. 

https://github.com/UCSD-E4E/engineering_leadership_workshops/commit/91aebf6d4496cc0ea0a7553d50ca8dc6e090d547